batch/batch/worker/worker.py

```diff
@@ -1002,8 +1052,56 @@ def container_is_running(self):
     def container_finished(self):
         return self.process is not None and self.process.returncode is not None
 
-    async def upload_log(self):
-        await self.worker.file_store.write_log_file(
+    async def get_log(self):
+        if os.path.exists(self.log_path):
+            stream = await self.fs.open(self.log_path)
+            async with stream:
+                return (await stream.read()).decode()
+        return ''
+
+    def __str__(self):
+        return f'container {self.name}'
+
+
+class Task(Container):
```

Comment:
I'm not a fan of this name. We already think of these as containers. Can we maybe just call this `LogUploadingContainer`? And then we don't need to change the name of `copy_container` etc.

Reply:
Alternatively, this entire class could just be a `Job` method:
```
def run_then_upload_log(self, c: Container):
    try:
        await c._run()
    finally:
        self._step(...):
            # ... copy the log ...
```

Reply:
Actually, I substantially prefer the method approach. Let's avoid subclasses whenever possible.